src/main/java/com/minecolonies/coremod/entity/ai/citizen/shepherd/EntityAIWorkShepherd.java

```diff
@@ -0,0 +1,297 @@
+          ,worker.getWorkBuilding().getLocation().getY(),worker.getWorkBuilding().getLocation().getZ()) > MAX_DISTANCE_FROM_HUT )
+        {
+            return START_WORKING;
+        }
+
+        job.setSheep(world.getEntitiesWithinAABB(
+          EntitySheep.class,
+          this.getTargetableArea(currentSearchDistance),
+          sheep ->
+          {
+              final double range = worker.getWorkBuilding().getLocation().getDistance(sheep.getPosition().getX(),
+                sheep.getPosition().getY(), sheep.getPosition().getZ());
+              return !sheep.getSheared() && range <= MAX_DISTANCE_FROM_HUT;
+          }
+        ));
+
+        worker.addExperience(EXP_PER_SHEEP);
+
+        if (!job.getSheep().isEmpty())
+        {
+            return SHEPHERD_WALK_TO_SHEEP;
+        }
+        else
+        {
+            currentSearchDistance++;
+        }
+        return SHEPHERD_SEARCH_FOR_SHEEP;
+    }
+
+    private AxisAlignedBB getTargetableArea(final double range)
+    {
+        return this.worker.getEntityBoundingBox().expand(range, HEIGHT_DETECTION_RANGE, range);
+    }
+
+    private AIState shearSheep()
+    {
+        final List<EntitySheep> sheepies = job.getSheep();
+
+        if (sheepies == null || sheepies.isEmpty())
+        {
+            return SHEPHERD_SEARCH_FOR_SHEEP;
+        }
+
+        if (checkForToolOrWeapon(ToolType.SHEARS))
+        {
+            return PREPARING;
+        }
+
+        final EntitySheep sheep = sheepies.get(0);
+
+        equipShears();
+
+        if (worker.getHeldItemMainhand() != null)
+        {
+            final List<ItemStack> items = sheep.onSheared(worker.getHeldItemMainhand(),
+              worker.worldObj,
+              worker.getPosition(),
+              net.minecraft.enchantment.EnchantmentHelper.getEnchantmentLevel(net.minecraft.init.Enchantments.FORTUNE, worker.getHeldItemMainhand()));
+
+            dyeSheepChance(sheep);
+
+            worker.getHeldItemMainhand().damageItem(1, worker);
+
+            for (ItemStack item : items)
+            {
+                worker.getInventoryCitizen().addItemStackToInventory(item);
+            }
+        }
+
+        if (sheep.getSheared())
+        {
+            sheepies.remove(sheep);
+            job.setSheep(sheepies);
+
+            if (job.getSheep().isEmpty())
+            {
+                return SHEPHERD_SEARCH_FOR_SHEEP;
+            }
+            return SHEPHERD_WALK_TO_SHEEP;
+        }
+        return SHEPHERD_SHEAR_SHEEP;
+    }
+
+    /**
+     *
+     */
+    private void dyeSheepChance(final EntitySheep sheep)
+    {
+        if (worker.getWorkBuilding() != null)
+        {
+            final int chanceToDye = worker.getWorkBuilding().getBuildingLevel();
+
+            final int rand = world.rand.nextInt(100);
+
+            System.out.println("chance: " + chanceToDye);
+            System.out.println("rand: " + rand);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS106)